src/praisonai-agents/praisonaiagents/goal/loop.py

```
@@ -0,0 +1,211 @@
+        """Load a persisted GoalState from the session store, if any."""
+        store = getattr(self, "_session_store", None) or getattr(
+            self, "session_store", None
+        )
+        if store is None:
+            return None
+        try:
+            session = store.get_session(
+                self._goal_state_key().split(":", 1)[1]
+            )
+            data = getattr(session, "metadata", {}).get("goal_state")
+        except Exception:  # pragma: no cover - persistence is best-effort
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)